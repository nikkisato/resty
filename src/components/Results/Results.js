import React from 'react';
import styles from './Results.css';
import PropTypes from 'prop-types';
import ReactJSON from 'react-json-view';
//This is going to display the results

export default function Results({ response }) {
  return (
    <>
      <h2>Results</h2>
      <div className={styles.results}>
        <ReactJSON src='response' />
      </div>
    </>
  );
}

Results.propTypes = {
  response: PropTypes.object
};
