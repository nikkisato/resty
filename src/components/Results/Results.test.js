import React from 'react';
import { shallow } from 'enzyme';
import Results from './Results';

describe('Result component', () => {
  it('match a snapshot', () => {
    const wrapper = shallow(<Results text='My Button' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match a snapshot with type prop passed', () => {
    const wrapper = shallow(<Results type='button' text='My Button' />);
    expect(wrapper).toMatchSnapshot();
  });
});
