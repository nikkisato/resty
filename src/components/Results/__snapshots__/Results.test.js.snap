// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component match a snapshot 1`] = `
<Fragment>
  <h2>
    Results
  </h2>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={true}
    displayObjectSize={true}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={4}
    name="root"
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={Object {}}
    theme="solarized"
    validationMessage="Validation Error"
  />
</Fragment>
`;

exports[`Result component match a snapshot with type prop passed 1`] = `
<Fragment>
  <h2>
    Results
  </h2>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={true}
    displayObjectSize={true}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={4}
    name="root"
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={Object {}}
    theme="solarized"
    validationMessage="Validation Error"
  />
</Fragment>
`;
